search/includes/classes/commands/class-corecommand.php

```diff
@@ -90,6 +91,35 @@ private function set_version( $indexable, $version ) {
 		}
 	}
 
+	/**
+	 * List all of the available indexes.
+	 *
+	 * @return array|WP_Error $indexes Array of available indexes.
+	 */
+	private function list_indexes() {
+		$path = '_cat/indices?format=json';
+
+		$response = Elasticsearch::factory()->remote_request( $path );
+		$body     = json_decode( wp_remote_retrieve_body( $response ) );
+
+		if ( ! is_array( $body ) ) {
+			$message = property_exists( $body, 'error' ) ? $body->error : 'Something went wrong during list_indexes().';
+			return new WP_Error( 'list-indexes-error', $message );
+		}
+
+		$indexes = array_column( $body, 'index' );
+		// Remove system operation indexes from list.
+		$indexes = array_filter( $indexes, function( $index ) {
+			if ( function_exists( 'str_starts_with' ) ) { // Only available for PHP 8+
+				return ! str_starts_with( $index, '.' );
+			}
+			return substr( $index, 0, 1 ) !== '.';
+		} );
```

Comment:
What do you think if we simplify this piece of code a bit?

```suggestion
		$indexes = array_filter( $indexes, fn( $index ) => ! $index || $index[0] !== '.' );
```

If `$index` cannot be empty, `fn( $index ) => $index[0] !== '.'`

Reply:
Sounds good, thank you!